jetstream/pre_treatment.py

```diff
@@ -15,6 +15,8 @@ class PreTreatment(ABC):
     calculating statistics.
     """
 
+    needs_analysis_period: bool = False
```

Comment:
Yea I went back and forth on those two options. I was trying to avoid passing an unnecessary parameter to every pre-treatment, but not sure this approach saves us anything.

Reply:
I guess the parameter wouldn't be exposed to users (as in they don't have to set it). Which would eliminate the `needs_analysis_period` parameter